soroban-env-host/src/storage.rs

```diff
@@ -78,14 +81,18 @@ impl Footprint {
         if let Some(existing) = self.0.get::<Rc<LedgerKey>>(key, budget)? {
             match (existing, ty.clone()) {
                 (AccessType::ReadOnly, AccessType::ReadOnly) => Ok(()),
-                (AccessType::ReadOnly, AccessType::ReadWrite) => {
-                    // The only interesting case is an upgrade
-                    // from previously-read-only to read-write.
+                (AccessType::ReadOnly, AccessType::ReadWrite)
```

Comment:
Hmm I think these aren't quite right. They might be, but .. I think in general commutative writes actually have to be _write-only_ a.k.a. "blind writes", read access to them should fail. In this case _possibly_ not because they're also supposed to be like .. a special kind of blind-write? One that carries a delta rather than a full value to write (because we're going to sum them) and the preimage going into the transaction is always zero for everyone.

(The point of write-only stuff is to avoid inconsistent reads between different partitions, which would make the true partitioning function observable, and it's not supposed to be.)

Reply:
Hmm that would be pretty restrictive when we don't need to be for this specific case. I guess not that restrictive since autobumps happen in core. Would it make more sense to name this AccessType `BumpWrite` (or some name that indicates it's just for bumps)?